rllib/tests/test_node_failure.py

```diff
@@ -0,0 +1,226 @@
+# This workload tests RLlib's ability to recover from failing workers nodes
+import time
+import unittest
+import threading
+from ray.exceptions import RayActorError
+
+import ray
+from ray.cluster_utils import Cluster
+from ray._private.test_utils import get_other_nodes
+
+from ray.rllib.algorithms.ppo import PPO, PPOConfig
+
+num_redis_shards = 5
+redis_max_memory = 10 ** 8
+object_store_memory = 10 ** 8
+num_nodes = 3
+
+message = (
+    "Make sure there is enough memory on this machine to run this "
+    "workload. We divide the system memory by 2 to provide a buffer."
+)
+assert (
+    num_nodes * object_store_memory + num_redis_shards * redis_max_memory
+    < ray._private.utils.get_system_memory() / 2
+), message
+
+
+class NodeFailureTests(unittest.TestCase):
+    def setUp(self):
+        # Simulate a cluster on one machine.
+        self.cluster = Cluster()
+
+        for i in range(num_nodes):
+            self.cluster.add_node(
+                redis_port=6379 if i == 0 else None,
+                num_redis_shards=num_redis_shards if i == 0 else None,
+                num_cpus=2,
+                num_gpus=0,
+                object_store_memory=object_store_memory,
+                redis_max_memory=redis_max_memory,
+                dashboard_host="0.0.0.0",
+            )
+        ray.init(address=self.cluster.address)
+
+    def tearDown(self):
+        ray.shutdown()
+
+    def test_fail_on_node_failure(self):
+        # We tolerate failing workers and don't stop training
+        config = (
+            PPOConfig()
+            .rollouts(
+                num_rollout_workers=6,
+                ignore_worker_failures=False,
```

Comment:
Wait, how do these settings "tolerate failing workers" as indicated in the comment above?

Shouldn't it be `ignore_worker_failures=True` or so?